firewood/src/v2/propose.rs

```diff
@@ -150,12 +150,12 @@ impl<T: api::DbView + Send + Sync> api::Proposal<T> for Proposal<T> {
         Ok(Proposal::new(ProposalBase::Proposal(self), data))
     }
 
-    async fn commit(self: Arc<Self>) -> Result<Arc<T>, api::Error> {
+    async fn commit(self: Arc<Self>) -> Result<(), api::Error> {
```

Comment:
Wasn't this supposed to return the latest view upon success? Now we have to go back to the database and get the latest view?

Reply:
simply a question, should not block merge

Reply:
The issue here is that T is specified by the caller. This really needs T specified by the implementation. It's difficult to implement it to return the specific type asked for by the caller, so that's why I got rid of T completely.

This can be done, but I don't think it's critical so I'm just removing the generic here as a start (you'd have to do that anyway).